ext/adaptors/ranges-lib.tex

```diff
@@ -192,7 +195,9 @@
     }
   public:
     constexpr bool empty() const requires ForwardRange<const D>;
-    constexpr explicit operator bool() const requires ForwardRange<const D>;
+    @\added{template <class R = D>}@
+      @\added{requires requires(const R\& r) \{ ranges::empty(r); \}}@
+    constexpr explicit operator bool() const @\removed{requires ForwardRange<const D>}@;
```

Comment:
I'd rather not.

Reply:
You prefer to keep it overconstrained? Why?

Reply:
(This change allows `subrange` to implement `empty` and get `operator bool` from `view_interface`)

Reply:
I know. And yes, I have a weak preference to keep it "overconstrained." The fact that some of the members must be templates is an ugly hack that's only necessary because concepts has no support for associated types. I don't want to see that ugly hack spread farther than is necessary.

Reply:
Correction: `template` is only needed here because of GCC 82507. [[temp.inst]/16](http://eel.is/c++draft/temp.inst#16) says I can write:
```c++
constexpr explicit operator bool() const requires requires { ranges::empty(derived()); }
```
and the constraints won't be checked until overload resolution time. Would that be acceptable?

Reply:
That's fine. I have no strong feelings about it now.